source/MultiGrid/MultiGrid.js

```diff
@@ -99,15 +100,15 @@ export default class MultiGrid extends Component {
     const { columnWidth, fixedColumnCount, fixedRowCount, rowHeight } = this.props
 
     if (
-      columnWidth !== nextProps.columnWidth ||
+      columnWidth instanceof Function || columnWidth !== nextProps.columnWidth ||
       fixedColumnCount !== nextProps.fixedColumnCount
     ) {
       this._leftGridWidth = null
     }
 
     if (
       fixedRowCount !== nextProps.fixedRowCount ||
-      rowHeight !== nextProps.rowHeight
+      rowHeight instanceof Function || rowHeight !== nextProps.rowHeight
```

Comment:
Users should call `recomputeGridSize` if something has happened to invalid one or more columns/rows. `Grid` then [resets its own cache when `recomputeGridSize` is called](https://github.com/bvaughn/react-virtualized/blob/master/source/Grid/Grid.js#L273-L276).